pkgs/development/libraries/openssl/default.nix

```diff
@@ -36,7 +36,9 @@ let
 
     outputs = [ "bin" "dev" "out" "man" ] ++ optional withDocs "doc";
     setOutputFlags = false;
-    separateDebugInfo = !(stdenv.hostPlatform.useLLVM or false) && stdenv.cc.isGNU;
+    separateDebugInfo = !(stdenv.hostPlatform.useLLVM or false)
+      && stdenv.hostPlatform == stdenv.buildPlatform # unable to fix infinite recursion on stdenv.cc
+      && stdenv.cc.isGNU;
```

Comment:
the fix is good, but it sounds like we need to tweak llvm to work with separate-debug-info.sh, since I bet that everything is broken, not just openssl.

Reply:
Right, that certainly seems preferable, but I have no idea about difficulty of that :-)